pkg/kexec/kexec_linux.go

```diff
@@ -1,15 +1,41 @@
-// Copyright 2015-2017 the u-root Authors. All rights reserved
+// Copyright 2015-2019 the u-root Authors. All rights reserved
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
 package kexec
 
 import (
 	"fmt"
+	"os"
 	"syscall"
 )
 
-// Reboot executes a kernel previously loaded with FileInit.
+// DefaultOpts returns the default options.
+func DefaultLinuxOpts() *LinuxOpts {
+	// TODO: specialize for each arch.
+	return &LinuxOpts{
+		PrependRSDP: true,
+		DTBPath:     "/sys/firmware/fdt",
+	}
+}
+
+// Opts contains options for the kexec on Linux.
+type LinuxOpts struct {
+	// The initramfs to pass to the next kernel. The caller is responsible for
+	// closing this file. This value is optional.
+	Initramfs *os.File
+
+	// The command line for the next kernel.
+	CmdLine string
+
+	// Set to true to prepend the RSDP to the command line. (amd64 only)
+	PrependRSDP bool
+
+	// Path to the device tree. (arm/arm64 only)
```

Comment:
honest, I'm convinced, the architecture qualifications are not needed. I also just realized this ought to work on riscv.